tests/integration/suite/test_app.py

```diff
@@ -647,6 +647,92 @@ def test_app_has_helmversion(admin_pc, admin_mc, remove_resource):
     assert app2.helmVersion == "helm_v3"
 
 
+def test_app_upgrade_has_helmversion(admin_pc, admin_mc, remove_resource):
+    """Test helm version exists on new chart versions when added to an
+    existing catalog"""
+    app_client = admin_pc.client
+    catalog_client = admin_mc.client
+    catalog_name = random_str()
+    app1_name = random_str()
+    app2_name = random_str()
+    cat_base = "catalog://?catalog=" + catalog_name +\
+        "&template=rancher-v3-issue&version="
+    helm3_catalog = catalog_client.create_catalog(
+      name=catalog_name,
+      branch="helmversion-onupdate-1v",
+      url="https://github.com/brendarearden/integration-test-charts",
+      helmVersion="helm_v3"
+    )
+    remove_resource(helm3_catalog)
+    wait_for_template_to_be_created(catalog_client, catalog_name)
+
+    ns = admin_pc.cluster.client.create_namespace(name=random_str(),
+                                                  projectId=admin_pc.
+                                                  project.id)
+    remove_resource(ns)
+    # creating app with existing chart version in catalog
+    app1 = app_client.create_app(
+        name=app1_name,
+        externalId=cat_base+"0.1.0&namespace="+ns.name,
+        targetNamespace=ns.name,
+        projectId=admin_pc.project.id,
+    )
+    remove_resource(app1)
+    wait_for_workload(app_client, ns.name, count=1)
+    app1 = app_client.reload(app1)
+    # check that the correct helm version is on the app
+    assert "helmVersion" in app1
+    assert app1.helmVersion == "helm_v3"
+    # changing branch on catalog to simulate adding a new chart version to the
+    # catalog
+    catalog_data = {
+      'name': catalog_name,
+      'branch': "helmversion-onupdate-2v",
+      'url': "https://github.com/brendarearden/integration-test-charts",
+      'helmVersion': "helm_v3"
+    }
+    helm3_catalog = catalog_client.update(helm3_catalog, catalog_data)
+
+    def ensure_updated_catalog(catalog):
+        catalog = catalog_client.reload(catalog)
+        templates = catalog_client.list_template(catalogId=catalog.id).data
+        templatesString = ','.join([str(i) for i in templates])
+        if "0.1.1" in templatesString:
+            return catalog
+        return None
+    helm3_catalog = wait_for(
+      lambda: ensure_updated_catalog(helm3_catalog),
+      fail_handler=lambda:
+      "Timed out waiting for catalog to stop transitioning")
+    project_client = user_project_client(admin_pc, admin_pc.project)
+    # update existing app with new version to ensure correct
```

Comment:
Ideally the template should be checked as that is really our source of truth, the app could be holding old data from pre upgrade thus giving a false positive.

Reply:
Both can be checked and this also tests the upgrade in general.